examples/spider/train_sql_agent_npu.py

```diff
@@ -0,0 +1,200 @@
+"""Train an SQL agent on the Spider dataset using Agent-lightning.
```

Comment:
what's the difference of this new script and the old script?

Reply:
We don't want to include a dedicated script in a CI-maintained examples for some hardwares that is not included in the CI., unless we will have a NPU machine for daily testing.